sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -665,6 +677,30 @@ def __init__(self, windowing):
     def expand(self, pcoll):
       return pcoll | self._window_transform
 
+    @staticmethod
+    def _parse_time_unit(value, name):
+      time_units = {
+          'ms': 0.001, 's': 1, 'm': 60, 'h': 60 * 60, 'd': 60 * 60 * 12
+      }
+      value, suffix = re.match(r'^(.*?)([^\d]*)$', str(value)).groups()
+      # Default to seconds if time unit suffix is not defined
+      if not suffix:
+        suffix = 's'
+      if not value:
+        raise ValueError(
+            f"Invalid windowing {name} value "
+            f"'{suffix if not value else value}'. "
+            f"Must provide numeric value.")
+      if suffix not in time_units:
+        raise ValueError((
+            "Invalid windowing {} time unit '{}'. " +
+            "Valid time units are {}.").format(
+                name,
+                suffix,
+                ', '.join("'{}'".format(k) for k in time_units.keys())))
```

Comment:
You could probably just do `', '.join(time_units.keys())`.

Reply:
I thought this looked prettier...

`Invalid windowing size time unit 't'. Valid time units are 'ms', 's', 'm', 'h', 'd'.`
vs.
`Invalid windowing size time unit 't'. Valid time units are ms, s, m, h, d`

Reply:
OK, that's fine.